### DISCUSSION TITLE
How do I update the old syntax `vim.diagnostic.config.float.close_events = { "User MyEvent" }`?

### DISCUSSION THREAD

**Discussion by nyngwang** [neovim/neovim#25809] [Q&A] [answered]
As title. I want to use a `User` event with the pattern `MyEvent` to be recognized by the `vim.diagnostic.config` settings. Thanks in advance!

To be more specific, I got this error after updating Neovim version to the current `HEAD`:

```
Error  15:17:06 msg_show.lua_error Error detected while processing CursorHold Autocommands for "*":
15:17:06 msg_show Error executing lua callback: /usr/local/share/nvim/runtime/lua/vim/lsp/util.lua:1490: Invalid 'event': 'User MurmurCloseDiag'
stack traceback:
	[C]: in function 'nvim_create_autocmd'
	/usr/local/share/nvim/runtime/lua/vim/lsp/util.lua:1490: in function 'close_preview_autocmd'
	/usr/local/share/nvim/runtime/lua/vim/lsp/util.lua:1685: in function 'open_floating_preview'
	/usr/local/share/nvim/runtime/lua/vim/diagnostic.lua:1465: in function 'open_float'
	/Users/ningwang/.config/nvim/lua/plugins/UX.lua:283: in function </Users/ningwang/.config/nvim/lua/plugins/UX.lua:277>
```

**Comment by zeertzjq** [comment]
Remove ` MyEvent` and just use `User`, because the pattern `MyEvent` never actually worked.

**Reply by nyngwang** [reply]
Thanks for your kindness. What if I just want to use `User` with the pattern `MyEvent` to trigger the close event?

**Reply by zeertzjq** [reply]
I don't think that has ever worked.

**Reply by nyngwang** [reply]
It used to be working. I added that line in my config but I didn't add a note about the source.

**Reply by zeertzjq** [reply, marked as answer]
No it didn't. It just behaved like a `User` autocommand with a `*` pattern without giving an error.

**Reply by nyngwang** [reply]
Is there a way to allow only one kind of `User` event, i.e. a `User` event with a custom pattern e.g. `MyEvent`, to trigger the close event of the floating window of `vim.diagnostic`?
